storage/innobase/trx/trx0trx.cc

```diff
@@ -918,6 +918,7 @@ trx_start_low(
 
 #ifdef WITH_WSREP
 	trx->xid.null();
+	trx->wsrep = wsrep_on(trx->mysql_thd);
```

Comment:
It would be good to add a comment why we do not trust this field in several places in `ha_innodb.cc`. If the field can’t be trusted in some cases, maybe it shouldn’t be trusted in any case and should be removed altogether, to avoid any inconsistencies? Or the initialization should be fixed.

Reply:
Can we remove the assignments to `trx->wsrep` from `innobase_trx_init()` and `innobase_commit_low()` and simply let the field be 0 when there is no active transaction?

Reply:
Removed all all assignments to `trx->wsrep` except the one in `trx_start_low()`.